dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -280,27 +281,22 @@ private CodeBlock generatePlainSerializer(
             Optional<ParameterEncoderType> maybeParameterEncoderType) {
         return type.match(new ArgumentType.Cases<>() {
             @Override
-            public CodeBlock primitive(TypeName _typeName, String parameterSerializerMethodName) {
+            public CodeBlock primitive(TypeName _typeName, String _parameterSerializerMethodName) {
                 return maybeParameterEncoderType.map(this::parameterEncoderType).orElseGet(() -> {
                     return CodeBlock.of(
-                            "$L.$L($S, $L.$L($L));",
-                            REQUEST,
-                            singleValueMethod,
-                            key,
-                            PARAMETER_SERIALIZER,
-                            parameterSerializerMethodName,
-                            argName);
+                            "$L.$L($S, $L);", REQUEST, singleValueMethod, key, generateSerializerCall(argName, type));
```

Comment:
This `generateSerializerCall` is weird. We're already matching on `type` above, so it's unusual to pass the full `type` into a helper method where we match on it again.

This feels like too much DRY - let's keep this as it was.

Reply:
Fixed